.github/workflows/deploy.yaml

```diff
@@ -20,7 +20,7 @@ jobs:
   deploy:
     name: Upload to Amazon S3
     runs-on: ubuntu-latest
-    if: ${{ github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags/v') }}
+    if: ${{ github.ref == 'refs/heads/master' || github.ref == 'refs/heads/testing' || startsWith(github.ref, 'refs/tags/v') }}
```

Comment:
we deploy from master on every change so i don't see how using `testing` branch would be different from using master branch?

also would deployment from testing branch be shown in https://ami.sageit.org/ ?  if so i'm not sure i like the idea of putting a bunch of testing AMIs on the same page as prod AMIs.  i think that might get kinda messy but i can be persuaded otherwise.

Reply:
the purpose of this PR is to enable pre-merge testing of packer changes, and I don't think iterating on those packer changes should require merging the intermediate change here.

I had no idea about ami.sageit.org, but I see there are already several test images listed (a few I built manually for testing, and a few from developing the docker notebook product). how do you feel about a change to the dashboard script to ignore AMIs with 'test' in the name?

Reply:
> the purpose of this PR is to enable pre-merge testing of packer changes, and I don't think iterating on those packer changes should require merging the intermediate change here.

yes, good point.  However I'm not sure why you even need to involve the CI deployment at all to accomplish pre-merge testing.  While in development, can you not just use your SSO account to deploy test AMIs to AWS and make them available for you to do pre-merge testing?


> how do you feel about a change to the dashboard script to ignore AMIs with 'test' in the name?

I like this idea.

Reply:
done in Sage-Bionetworks-IT/imagecentral-dashboard#11